src/models/parentModel.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require('mongoose');
+
+const parentSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: [true, 'Please provide name']
+    },
+    userName: {
+        type: String,
+        required: [true, 'Please provide user name']
+    },
+    email: {
+        type: String,
+        required: [true, 'Please provide email address'],
+        unique: true, 
+        lowercase: true,
+        match: [/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/, 'Please provide a valid email']
+    },
+    password: {
+        type: String,
+        required: [true, 'Please provide a password'],
+        minLength: [6, 'Password must be at least 6 characters long'],
```

Comment:
please remember that this value will be the hashed value of the raw text entered

Reply:
the length validation can that be done with a validator at the controller/middleware level